gradle/gradle-mvn-push.gradle

```diff
@@ -31,112 +14,67 @@ def getSnapshotRepositoryUrl() {
   return findProperty("SNAPSHOT_REPOSITORY_URL") ?: "https://oss.jfrog.org/artifactory/oss-snapshot-local"
 }
 
-def getRepositoryUsername() {
-  return findProperty("BINTRAY_USER") ?: "no.bintray.user"
+task sourcesJar(type: Jar, dependsOn: classes) {
+  archiveClassifier = "sources"
+  from sourceSets.main.allSource, "build/generated/source/kapt/main", "build/generated/source/kaptKotlin/main"
```

Comment:
Sources can also be in:
```groovy
'build/generated/source/kapt/main',
'build/generated/source/kapt/debug',
'build/generated/source/kapt/release',
'build/generated/source/kaptKotlin/main',
'build/generated/source/kaptKotlin/debug',
'build/generated/source/kaptKotlin/release',
'build/tmp/kapt/main/kotlinGenerated'
```

(Given [Arrow docs](https://github.com/arrow-kt/arrow#additional-setup)).

Reply:
I thought that was only needed when the project uses the macro annotations, isn't it?

Reply:
That's where the generated code lives, depending on which variant of the project you've compiled.  So regarding that, you'd probably need to think on which variants you'd compile for deploys so you don't miss the sources.